packages/react-jss/src/withStyles.js

```diff
@@ -142,7 +143,8 @@ export default function withStyles<Theme: {}, S: Styles<Theme>>(
           ...contextSheetOptions,
           index,
           meta: `${displayName}, ${isThemingEnabled ? 'Themed' : 'Unthemed'}, Static`,
-          classNamePrefix: this.classNamePrefix
+          classNamePrefix: this.classNamePrefix,
+          link: getObservableStyles(themedStyles) !== null
```

Comment:
Now I think because of the separation in observable and function styles, we have double overhead for those 2 function calls.

Reply:
We could have one function and do it in one go.

Reply:
Also we don't need getObservableStyles, we need hasObservables() from the logic above. It could be potentially one `getDynamicStyles() : {styles, hasObservables, hasFunctions}`

Reply:
I think this way it's a lot cleaner and the core doesn't need to know about observables or functions in that matter.

Reply:
It is, but performance is the primary priority here. I think iterating over styles is quite a bit of overhead.

Reply:
I think we can't evaluate that, because most people don't know this API exist. We need to promote it more, but in the end, we don't know how and were it is used.

Observables are part of the default preset, so it would be not logical if they didn't work with react-jss, which states using default preset.

Also even if we would remove observables from a default preset, it wouldn't be logical if react-jss doesn't work when user defined a new jss instance with an observable plugin.

Reply:
I think we just need to figure out a good way to use observables with react-jss and get some use cases we can target for.